src/components/Input/index.ts

```diff
@@ -0,0 +1,45 @@
+import Block from '../../utils/Block';
+import template from './input.hbs';
+import validateValueByName from '../../utils/Validate';
+import * as styles from './input.module.scss';
+
+interface InputProps {
+  label?: string;
+  type?: string;
```

Comment:
Если type принимает ограниченное количество значений, то лучше использовать здесь не string, а discriminated union
```suggestion
  type?: 'text' | 'number' и тд;
```